packages/core/src/components/essentials/tooltip.ts

```diff
@@ -26,26 +29,42 @@ export class Tooltip extends Component {
 		this.init();
 	}
 
-	init() {
-		// Grab the tooltip element
-		const holder = select(this.services.domUtils.getHolder());
-		const chartprefix = Tools.getProperty(
-			this.model.getOptions(),
-			"style",
-			"prefix"
-		);
-		this.tooltip = DOMUtils.appendOrSelect(
-			holder,
-			`div.${settings.prefix}--${chartprefix}--tooltip`
-		);
+	handleShowTooltip = (e) => {
+		const data = e.detail.data || e.detail.items;
+		const defaultHTML = this.getTooltipHTML(e);
 
-		// Apply html content to the tooltip
-		const tooltipTextContainer = DOMUtils.appendOrSelect(
-			this.tooltip,
-			"div.content-box"
-		);
-		this.tooltip.style("max-width", null);
+		// if there is a provided tooltip HTML function call it
+		if (
+			Tools.getProperty(
+				this.model.getOptions(),
+				"tooltip",
+				"customHTML"
+			)
+		) {
+			this.tooltipTextContainer.html(
```

Comment:
`tooltipTextContainer` seems to be only used here, so why would we initialize it into `this` inside `render`? we should just move it here

Reply:
In order to use named function when adding event listener, I have to put `this.handleShowTooltipVar` as the second parameter of `this.services.events.addEventListener()`. I can't use the following way to add event listener hence I still need `this.tooltipTextContainer` for accessing it in `this.handleShowTooltip` function and then assigned `this.handleShowTooltip` function to a named function `this.handleShowTooltipVar` without passing `e` as parameter to `this.handleShowTooltipVar` when calling it in `this.services.events.addEventListener`.

```
		this.services.events.addEventListener(
			Events.Tooltip.SHOW,
			function(e){
                                ...
                                ...
                        }
		);
```

Reply:
`handleShowTooltip` can itself act as that function, you don't need an extra function to wrap `handleShowTooltip`. Please take a look at some of the graph classes that we have where we do the exact same thing

![image](https://user-images.githubusercontent.com/14989804/93099110-a5b60f00-f675-11ea-87c0-79b2eff7810c.png)

![image](https://user-images.githubusercontent.com/14989804/93099131-ae0e4a00-f675-11ea-9bcd-bffaa246cd0e.png)

There's really no reason for `handleShowTooltipVar`